app/src/org/commcare/tasks/ConnectionDiagnosticTask.java

```diff
@@ -67,15 +81,19 @@ public ConnectionDiagnosticTask(Context c) {
         TAG = ConnectionDiagnosticTask.class.getSimpleName();
     }
 
+    public void setListener(ConnectionDiagnosticListener<R> listener) {
+        this.listener = listener;
+    }
+
     @Override
-    protected Test doTaskBackground(Void... params) {
-        Test out = null;
+    protected NetworkState doTaskBackground(Void... params) {
+        NetworkState out = NetworkState.CONNECTED;
         if (!isOnline(this.c)) {
-            out = Test.isOnline;
+            out = NetworkState.DISCONNECTED;
         } else if (!pingSuccess(googleURL)) {

```

Comment:
So not getting a 200 response from a url doesn't mean that it's a captive portal but just that network is unavailable. Similarly getting 200 response doesn't mean that user is connected but the response might be supplied through captive portal. We might wanna try https://stackoverflow.com/a/14030276